pkg/services/rpcsrv/server_test.go

```diff
@@ -2003,6 +2003,107 @@ func TestSubmitNotaryRequest(t *testing.T) {
 	})
 }
 
+func TestGetRawNotaryPoolAndGetRawNotaryRequest(t *testing.T) {
+	var notaryRequest *payload.P2PNotaryRequest
+	rpcPool := `{"jsonrpc": "2.0", "id": 1, "method": "getrawnotarypool", "params": %s}`
+	rpcRequest := `{"jsonrpc": "2.0", "id": 1, "method": "getrawnotaryrequest", "params": ["%s", %d]}`
+
+	t.Run("disabled P2PSigExtensions", func(t *testing.T) {
+		chain, rpcSrv, httpSrv := initClearServerWithCustomConfig(t, func(c *config.Config) {
+			c.ProtocolConfiguration.P2PSigExtensions = false
+		})
+		defer chain.Close()
+		defer rpcSrv.Shutdown()
+		t.Run("getrawnotarypool", func(t *testing.T) {
+			body := doRPCCallOverHTTP(fmt.Sprintf(rpcPool, "[]"), httpSrv.URL, t)
+			checkErrGetResult(t, body, true, neorpc.InternalServerErrorCode)
+		})
+		t.Run("getrawnotaryrequest", func(t *testing.T) {
+			body := doRPCCallOverHTTP(fmt.Sprintf(rpcRequest, " ", 1), httpSrv.URL, t)
+			checkErrGetResult(t, body, true, neorpc.InternalServerErrorCode)
+		})
+	})
+
+	chain, rpcSrv, httpSrv := initServerWithInMemoryChainAndServices(t, false, true, false)
+	defer chain.Close()
+	defer rpcSrv.Shutdown()
+
+	submitNotaryRequest := func(t *testing.T, fail bool, errCode int64, params ...string) func(t *testing.T) {
+		return func(t *testing.T) {
+			rpcSubmit := `{"jsonrpc": "2.0", "id": 1, "method": "submitnotaryrequest", "params": %s}`
+			ps := `[` + strings.Join(params, ",") + `]`
+			req := fmt.Sprintf(rpcSubmit, ps)
+			body := doRPCCallOverHTTP(req, httpSrv.URL, t)
+			checkErrGetResult(t, body, fail, errCode)
+		}
+	}
+
+	compareTx := func(t *testing.T, expectedTx, actualTx transaction.Transaction) func(t *testing.T) {
+		return func(t *testing.T) {
+			require.Equal(t, expectedTx.Hash(), actualTx.Hash())
+			require.Equal(t, expectedTx.Nonce, actualTx.Nonce)
+			require.Equal(t, expectedTx.Sender(), actualTx.Sender())
+			require.Equal(t, expectedTx.Signers, actualTx.Signers)
+			require.Equal(t, expectedTx.ValidUntilBlock, actualTx.ValidUntilBlock)
+			require.Equal(t, expectedTx.Script, actualTx.Script)
+			require.Equal(t, expectedTx.Attributes, actualTx.Attributes)
+		}
+	}
```

Comment:
You don't need this function. Just use `tx.Hash()` and `tx.Size()` for expected/actual transactions to properly fill the cache in. After that use `require.Equal(t, expected, actual)`.

Reply:
Fixed.